### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by yang** [metaes/metaes#33] [Q&A] [answered]
Thank you for sharing this library!

Would you happen to have handy any rough ballparks around what to expect performance-wise? Thanks!

**Comment by bartosz-k** [comment, marked as answer]
Performance of MetaES is not something to be interested in to be honest. Even if it can be faster by doing some optimizations, compilations etc, you can immediately make is slow again by running MetaES inside MetaES to get more control instead.
 
Having said that, you can make many kinds of measurements, for example:

Native node.js:
> % time node -e 'let r=[]; for(let i=0;i<1_000_000;i++) {r.push(i)};'
node -e 'let r=[]; for(let i=0;i<1_000_000;i++) {r.push(i)};'  0,04s user 0,01s system 88% cpu 0,061 total

node.js running MetaES:
> % time node lib/repl -e 'let r=[]; for(let i=0;i<1_000_000;i++) {r.push(i)};'
node lib/repl -e 'let r=[]; for(let i=0;i<1_000_000;i++) {r.push(i)};'  1,39s user 0,03s system 104% cpu 1,350 total

**Comment by bartosz-k** [comment]
One more thing: you can always keep performance critical code as the native JavaScript and call it from MetaES. This way you loose introspection in this part, but still have the performance which may be crucial to keep some piece of software usable.

**Reply by bartosz-k** [reply]
@yang does it help?

**Reply by yang** [reply]
Very helpful! Thank you.
